### PR TITLE
Fix to set content-type correctly for gRPC Trailers-Only

### PR DESCRIPTION
**Pull request by minwoox**
Motivation:
We hardcoded the content-type with `application/grpc+proto` when sending gRPC Trailers-Only.
It is no harm to set the value because there's no following messages.
But I think it's better to set the content-type of the corresponding gRPC request so that a user
doesn't get surprised.

Modifications:
- Set the content-type of the corresponding gRPC request to the gRPC Trailers-Only.

Result:
- gRPC Trailers-Only now has the content-type of the corresponding gRPC request.

### CODE REVIEW

**Comment by trustin** [comment]
I see a build timeout - is there a chance our test hung?

**Comment by minwoox** [comment]
> I see a build timeout - is there a chance our test hung?

😱  Let me see what I missed.

File: grpc/src/main/java/com/linecorp/armeria/server/grpc/ArmeriaServerCall.java [hunk 1]
```diff
@@ -515,22 +517,22 @@ private void setClientStreamClosed() {
 
     // Returns ResponseHeaders if headersSent == false or HttpHeaders otherwise.
     static HttpHeaders statusToTrailers(
-            ServiceRequestContext ctx, Status status, Metadata metadata, boolean headersSent) {
-        final HttpHeadersBuilder trailers = GrpcTrailersUtil.statusToTrailers(
-                status.getCode().value(), status.getDescription(), headersSent);
+            ServiceRequestContext ctx, HttpHeadersBuilder trailerBuilder, Status status, Metadata metadata) {
```
[the remark this answers is not in the document]
Comment (minwoox) [reply grpc/src/main/java/com/linecorp/armeria/server/grpc/ArmeriaServerCall.java]: Fixed. 😉

**Comment by minwoox** [comment]
Build two more times and they all passed. 🤔

**Comment by minwoox** [comment]
Thank you reviewers!

**Pull request merged**